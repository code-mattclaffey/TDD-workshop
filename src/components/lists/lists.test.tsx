import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { List, ListItem } from '.';

afterEach(() => {
    cleanup();
});

describe('<List />', () => {
    it('should match inline snapshot', () => {
        const { container } = render(
            <List inline>
                <ListItem>List item</ListItem>
                <ListItem>List item</ListItem>
                <ListItem>List item</ListItem>
            </List>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match stacked snapshot', () => {
        const { container } = render(
            <List stacked>
                <ListItem>List item</ListItem>
                <ListItem>List item</ListItem>
                <ListItem>List item</ListItem>
            </List>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match ol snapshot', () => {
        const { container } = render(
            <List as="ol" stacked>
                <ListItem>List item</ListItem>
                <ListItem>List item</ListItem>
                <ListItem>List item</ListItem>
            </List>,
        );

        expect(container).toMatchSnapshot();
    });
});

describe('<ListItem />', () => {
    it('should match snapshot', () => {
        const { container } = render(<ListItem>List item</ListItem>);

        expect(container).toMatchSnapshot();
    });

    it('should match highlighted snapshot', () => {
        const { container } = render(
            <ListItem highlight>List item</ListItem>,
        );

        expect(container).toMatchSnapshot();
    });
});
