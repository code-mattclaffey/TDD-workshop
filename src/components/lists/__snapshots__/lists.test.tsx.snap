// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match inline snapshot 1`] = `
<div>
  <ul
    class="e-list e-list--inline"
  >
    <li
      class="e-list__item"
    >
      List item
    </li>
    <li
      class="e-list__item"
    >
      List item
    </li>
    <li
      class="e-list__item"
    >
      List item
    </li>
  </ul>
</div>
`;

exports[`<List /> should match ol snapshot 1`] = `
<div>
  <ol
    class="e-list e-list--stacked"
  >
    <li
      class="e-list__item"
    >
      List item
    </li>
    <li
      class="e-list__item"
    >
      List item
    </li>
    <li
      class="e-list__item"
    >
      List item
    </li>
  </ol>
</div>
`;

exports[`<List /> should match stacked snapshot 1`] = `
<div>
  <ul
    class="e-list e-list--stacked"
  >
    <li
      class="e-list__item"
    >
      List item
    </li>
    <li
      class="e-list__item"
    >
      List item
    </li>
    <li
      class="e-list__item"
    >
      List item
    </li>
  </ul>
</div>
`;

exports[`<ListItem /> should match snapshot 1`] = `
<div>
  <li
    class="e-list__item"
  >
    List item
  </li>
</div>
`;
